internal/controllers/topology/cluster/desired_state.go

```diff
@@ -511,7 +531,8 @@ func computeMachineDeployment(_ context.Context, s *scope.Scope, desiredControlP
 			desiredMachineDeploymentObj,
 			selectorForMachineDeploymentMHC(desiredMachineDeploymentObj),
 			s.Current.Cluster.Name,
-			machineDeploymentBlueprint.MachineHealthCheck)
+			machineDeploymentBlueprint.MachineHealthCheck,
+			currentMachineDeployment.MachineHealthCheck)
```

Comment:
`currentMachineDeployment.MachineHealthCheck)` may panic due to `currentMachineDeployment being nil.